xla/service/cpu/tests/onednn_memory_util_test.cc

```diff
@@ -0,0 +1,109 @@
+/* Copyright 2024 The OpenXLA Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#if defined(INTEL_MKL) && defined(ENABLE_ONEDNN_V3)
+
+#include "xla/service/cpu/onednn_memory_util.h"
+
+#include <string>
+
+#include "llvm/ADT/ArrayRef.h"
+#include "llvm/IR/Argument.h"
+#include "llvm/IR/BasicBlock.h"
+#include "llvm/IR/DerivedTypes.h"
+#include "llvm/IR/Function.h"
+#include "llvm/IR/IRBuilder.h"
+#include "llvm/IR/Type.h"
+#include "llvm/IR/Value.h"
+#include "xla/service/llvm_ir/llvm_util.h"
+#include "xla/shape.h"
+#include "xla/shape_util.h"
+#include "xla/test.h"
+#include "xla/tests/filecheck.h"
+#include "xla/tsl/lib/core/status_test_util.h"
+#include "tsl/platform/test.h"
+
+namespace xla {
+namespace cpu {
+namespace {
+
+class MemoryUtilTest : public ::testing::Test,
+                       public ::testing::WithParamInterface<
+                           std::tuple<PrimitiveType, std::vector<int64_t>>> {
+ protected:
+  constexpr static const char* pad_test_pattern_ = R"(
+    CHECK: %[[mref0:[0-9]+]] = insertvalue
+    CHECK: %[[mref1:[0-9]+]] = insertvalue
+    CHECK-SAME: [[arr:\[12 x i64\]]], [[arr]] } %[[mref0]], i64 255, 3
```

Comment:
Nit: We don't need to capture `mref1` and `arr`. 
```suggestion
    CHECK: %[[mref0:[0-9]+]] = insertvalue
    CHECK: %{{[0-9]+}} = insertvalue
    CHECK-SAME: [12 x i64], [12 x i64] } %[[mref0]], i64 255, 3
```

Reply:
Modified the filecheck pattern to match the new tests.